e2e/app/aou-elements/xpath-defaults.ts

```diff
@@ -0,0 +1,70 @@
+import TextOptions from './text-options';
+
+function textXpathHelper(opts: TextOptions) {
+  let txt;
+  if (opts.text) {
+    txt = `text()="${opts.text}" or @aria-label="${opts.text}" or @placeholder="${opts.text}"`;
+  } else if (opts.textContains) {
+    txt = `contains(text(),"${opts.textContains}") or contains(@aria-label,"${opts.textContains}") or contains(@placeholder,"${opts.textContains}")`;
+  } else if (opts.normalizeSpace) {
+    txt = `contains(normalize-space(), "${opts.normalizeSpace}")`;
+  }
```

Comment:
nit: else throw an error?

Reply:
not needed because textOptions are predefined, use un-predefined string leads to compile error.

Reply:
I don't totally follow what you're saying, but I don't think this is correct. Clearly I could pass an empty TextOptions here. The compiler would not fail.